things/api/things/http/transport.go

```diff
@@ -393,26 +393,48 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 	case errors.Error:
 		w.Header().Set("Content-Type", contentType)
 		switch {
+		case errors.Contains(errorVal, things.ErrUnauthorizedAccess):
+			w.WriteHeader(http.StatusUnauthorized)
+
+		case errors.Contains(errorVal, errInvalidQueryParams):
+			w.WriteHeader(http.StatusBadRequest)
+		case errors.Contains(errorVal, errUnsupportedContentType):
+			w.WriteHeader(http.StatusUnsupportedMediaType)
+
 		case errors.Contains(errorVal, things.ErrMalformedEntity):
 			w.WriteHeader(http.StatusBadRequest)
-		case errors.Contains(errorVal, things.ErrUnauthorizedAccess):
-			w.WriteHeader(http.StatusForbidden)
 		case errors.Contains(errorVal, things.ErrNotFound):
 			w.WriteHeader(http.StatusNotFound)
 		case errors.Contains(errorVal, things.ErrConflict):
+			w.WriteHeader(http.StatusConflict)
+		case errors.Contains(errorVal, things.ErrScanMetadata):
 			w.WriteHeader(http.StatusUnprocessableEntity)
-		case errors.Contains(errorVal, errUnsupportedContentType):
-			w.WriteHeader(http.StatusUnsupportedMediaType)
-		case errors.Contains(errorVal, errInvalidQueryParams):
+		case errors.Contains(errorVal, things.ErrSelectEntity):
+			w.WriteHeader(http.StatusUnprocessableEntity)
+		case errors.Contains(errorVal, things.ErrEntityConnected):
+			w.WriteHeader(http.StatusUnprocessableEntity)
+
+		case errors.Contains(errorVal, things.ErrCreateEntity):
 			w.WriteHeader(http.StatusBadRequest)
-		case errors.Contains(errorVal, things.ErrRemoveThing):
-			w.WriteHeader(http.StatusInternalServerError)
-		case errors.Contains(errorVal, things.ErrRemoveChannel):
-			w.WriteHeader(http.StatusInternalServerError)
+		case errors.Contains(errorVal, things.ErrUpdateEntity):
+			w.WriteHeader(http.StatusBadRequest)
+		case errors.Contains(errorVal, things.ErrViewEntity):
+			w.WriteHeader(http.StatusBadRequest)
+		case errors.Contains(errorVal, things.ErrRemoveEntity):
+			w.WriteHeader(http.StatusBadRequest)
+		case errors.Contains(errorVal, things.ErrConnect):
+			w.WriteHeader(http.StatusBadRequest)
+		case errors.Contains(errorVal, things.ErrDisconnect):
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
Use the multiple statement: 
```
		case errors.Contains(errorVal, things.ErrUpdateEntity),
                         errors.Contains(errorVal, things.ErrViewEntity),
                          ...
                         errors.Contains(errorVal, things.ErrRemoveEntity):
                        w.WriteHeader(http.StatusBadRequest)
```

Reply:
Be more precise, please.

Reply:
```
case errors.Contains(errorVal, things.ErrUpdateEntity),
  errors.Contains(errorVal, things.ErrViewEntity),
  errors.Contains(errorVal, things.ErrRemoveEntity),
  errors.Contains(errorVal, things.ErrConnect),
  errors.Contains(errorVal, things.ErrDisconnect):
  w.WriteHeader(http.StatusBadRequest)
```

Reply:
Ok. The slight problem I see is where I want to have a semantic grouping which collides with this nice syntactic feature. In that case, I would prefer longer code and semantic grouping.